src/emulate.c

```diff
@@ -665,30 +666,57 @@ static inline bool op_jal(struct riscv_t *rv, uint32_t insn)
     return false;
 }
 
+static inline int64_t time_us(void)
```

Comment:
Ditto. Don't return the copy of `struct timeval` object. Instead, use pointers explicitly.